File: 1021.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        vector<char> stack;
        string ans;
        
        if(s.size()==1)
        {return "";}
        
        stack.push_back(s[0]);
        for(int i=1;i<s.length();i++)
        {
            if(s[i]=='(')
            {
               if(!stack.empty())
               {
                   ans.push_back(s[i]);
               }
               stack.push_back(s[i]); 
            }
            else
            {
                if(!(stack.size()==1))
                {
                    ans.push_back(s[i]);
                }
                stack.pop_back();
            }
        }
        
        return ans;
    }
};

File: 1021.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        stack=list()
        s=list(s)
        eraseind=list()
        startind=0
        endind=0

        # 지워야할 인덱스 찾는 과정
        for i in range(len(s)):
            if s[i]=='(':
                if len(stack)==0:
                    startind=i    
                stack.append('(')
            else:
                    stack.pop()
                    if(len(stack)==0):
                        endind=i
                        eraseind.append(startind)
                        eraseind.append(endind)
                        
        # 지워야할 인덱스 지워주기     
        for i in range(len(eraseind)):
            del s[eraseind[i]-i]
            
        return "".join(s)       
